storage/allocator.go

```diff
@@ -244,6 +244,57 @@ func (a *allocator) RebalanceTarget(required proto.Attributes, existing []proto.
 	return s
 }
 
+// RemoveTarget returns a suitable replica to remove from the provided replica
+// set. It attempts to consider which of the provided replicas would be the best
+// candidate for removal.
+func (a *allocator) RemoveTarget(existing []proto.Replica) (proto.Replica, error) {
```

Comment:
Add a TODO here. You're going to need to supply a slice of Attributes from the zone config. Based on that, you'll determine which replicas are appropriate to remove: first based on attributes, then based on the store statistics.

As a simple example, if the zone config requires three replicas:

  [Attributes([]string{"us-east-1a"}), Attributes([]string{"us-east-1b"}), Attributes([]string{"us-east-1c"})]

And you have one replica with "us-east-1a", one with "us-east-1b", and two with "us-east-1c", you're only going to want to consider the two "us-east-1c" replicas as targets.